src/lib/address_resolve/AddressResolve_DefaultImpl.cpp

```diff
@@ -112,6 +100,65 @@ NodeLookupAction NodeLookupHandle::NextAction(System::Clock::Timestamp now)
     return NodeLookupAction::KeepSearching();
 }
 
+bool NodeLookupResults::UpdateResults(const ResolveResult & result)
+{
+    auto newScore = Dnssd::IPAddressSorter::ScoreIpAddress(result.address.GetIPAddress(), result.address.GetInterface());
+
+    uint8_t insertAtIndex = 0;
+    for (; insertAtIndex < kNodeLookupResultsLen; insertAtIndex++)
+    {
+        if (insertAtIndex >= count)
+        {
+            // This is a new entry.
+            break;
+        }
+
+        auto & oldAddress = results[insertAtIndex].address;
+        auto oldScore     = Dnssd::IPAddressSorter::ScoreIpAddress(oldAddress.GetIPAddress(), oldAddress.GetInterface());
+        if (newScore > oldScore)
+        {
+            // This is a score update, it will replace a previous entry.
+            break;
+        }
+    }
+
+    if (insertAtIndex == kNodeLookupResultsLen)
+    {
+        return false;
+    }
+
+    // Move the following valid entries one level down.
+    for (auto i = count; i > insertAtIndex; i--)
+    {
+        if (i >= kNodeLookupResultsLen)
+        {
+            continue;
+        }
+
+        results[i] = results[i - 1];
+    }
+
+    // If the number of valid entries is less than the size of the array there is an additional entry.
+    if (count < kNodeLookupResultsLen)
+    {
+        count++;
+    }
+
+    auto & updatedResult = results[insertAtIndex];
+    updatedResult        = result;
+    if (!updatedResult.address.GetIPAddress().IsIPv6LinkLocal())
+    {
+        // Only use the DNS-SD resolution's InterfaceID for addresses that are IPv6 LLA.
```

Comment:
This seems like it maybe should be common logic that lives in the resolver itself?

Reply:
I am not disagreeing but I'm not sure if some implementation needs to not filter that and that is why it lives in the "DefaultImpl" that can be overridden.

Reply:
"the resolver" is actually multiple implementations, so in some ways this is the right place to have the logic if it's going to only live in one place....  Note also that this is pre-existing.